internal/database/client.go

```diff
@@ -0,0 +1,50 @@
+package database
+
+import (
+	"fmt"
+
+	"github.com/boltdb/bolt"
+)
+
+const (
+	databaseFile = "madprobe.db"
+	probeBucket  = "Probe"
+)
+
+type client struct {
+	db *bolt.DB
+}
+
+var c *client
+
+func GetClient() (*client, error) {
+	if c != nil {
+		return c, nil
+	}
+	var err error
+	c, err = newClient()
+	return c, err
+}
+
+func newClient() (*client, error) {
```

Comment:
Since we are only using one specific database here, I do not think adding a new struct is really necessary. In the future, if there is a need of handling other persistence solution, I'll guess we come back to a mix of this and interfaces. For now, no need to be "complicated".
```suggestion
func newClient() (*bolt.DB, error) {
```

Reply:
This has been simplified, there is still a client used to contain the map of probes, is it fine like this ?